orangecontrib/text/topics/topics.py

```diff
@@ -0,0 +1,133 @@
+from gensim import matutils
+import numpy as np
+from gensim.corpora import Dictionary
+
+from Orange.data import StringVariable, ContinuousVariable, Domain
+from Orange.data.table import Table
+from orangecontrib.text.corpus import Corpus
+
+
+def chunks(iterable, chunk_size):
+    chunk = []
+    for item in iterable:
+        chunk.append(item)
+        if len(chunk) == chunk_size:
+            yield chunk
+            chunk = []
+    if chunk:
+        yield chunk
+
+
+MAX_WORDS = 1000
+
+
+class Topics(Table):
+    """ Dummy wrapper for Table so signals can distinguish Topics from Data.
+    """
+
+    def __new__(cls, *args, **kwargs):
+        """ Bypass Table.__new__. """
+        return object.__new__(Topics)
+
+
+class GensimWrapper:
+    name = NotImplemented
+    Model = NotImplemented
+    num_topics = NotImplemented
+
+    def __init__(self, **kwargs):
+        for k, v in kwargs.items():
+            setattr(self, k, v)
+        self.kwargs = kwargs
+        self.model = None
+        self.topic_names = []
+
+    def fit(self, corpus, progress_callback=None):
+        """ Train the model with the corpus.
+
+        Args:
+            corpus (Corpus): A corpus to lear topics from.
+        """
+        if not len(corpus.dictionary):
+            return None
+        self.reset_model(corpus)
+        self.update(corpus, progress_callback)
+        self.topic_names = ['Topic{} ({})'.format(i, ', '.join(words))
+                            for i, words in enumerate(self._topics_words(3), 1)]
+
+    def dummy_method(self, *args, **kwargs):
+        pass
+
+    def reset_model(self, corpus):
+        # prevent model from updating
+        _update = self.Model.update
+        self.Model.update = self.dummy_method
+        self.id2word = Dictionary(corpus.ngrams, prune_at=None)
+        self.model = self.Model(corpus=corpus,
```

Comment:
It's one of the gensim particularities. Sometimes (LDA, LSI) you can pass None  and sometimes not (HDP).